src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -91,101 +92,82 @@ public override bool TryCreateColumnContent(string columnName, out FrameworkElem
+                    // solution is never supposed to be kept alive for long time, meaning there is bunch of conditional weaktable or weak reference
+                    // keyed by solution/project/document or corresponding states. this will cause all those to be kept alive in memory as well.
+                    // probably we need to dig in to see how expensvie it is to support this
+                    var controlService = _excerptResult.Document.Project.Solution.Workspace.Services.GetService<IContentControlService>();
+                    controlService.AttachToolTipToControl(content, () =>
+                        CreateDisposableToolTip(_excerptResult.Document, _excerptResult.Span));
 
                     return true;
                 }
 
                 return false;
             }
 
-            private DisposableToolTip CreateDisposableToolTip()
+            private DisposableToolTip CreateDisposableToolTip(Document document, TextSpan sourceSpan)
             {
                 Presenter.AssertIsForeground();
 
-                // Create a new buffer that we'll show a preview for.  We can't search for an 
-                // existing buffer because:
-                //   1. the file may not be open.
-                //   2. our results may not be in sync with what's actually in the editor.
-                var textBuffer = CreateNewBuffer();
+                var controlService = document.Project.Solution.Workspace.Services.GetService<IContentControlService>();
+                var sourceText = document.GetTextSynchronously(CancellationToken.None);
 
-                // Create the actual tooltip around the region of that text buffer we want to show.
-                var toolTip = new ToolTip
+                var excerptService = document.Services.GetService<IDocumentExcerptService>();
+                if (excerptService != null)
                 {
-                    Content = CreateToolTipContent(textBuffer),
-                    Background = (Brush)Application.Current.Resources[EnvironmentColors.ToolWindowBackgroundBrushKey]
-                };
-
-                // Create a preview workspace for this text buffer and open it's corresponding 
-                // document.  That way we'll get nice things like classification as well as the
-                // reference highlight span.
-                var newDocument = Document.WithText(textBuffer.AsTextContainer().CurrentText);
-                var workspace = new PreviewWorkspace(newDocument.Project.Solution);
-                workspace.OpenDocument(newDocument.Id);
-
-                return new DisposableToolTip(toolTip, workspace);
+                    var excerpt = Presenter.ThreadingContext.JoinableTaskFactory.Run(() => excerptService.TryExcerptAsync(document, sourceSpan, ExcerptMode.Tooltip, CancellationToken.None));
+                    if (excerpt != null)
```

Comment:
does the default behavior which is showing data from Roslyn Document.

Reply:
in general, if there is a question in the PR that relates to a lack of clarity, it would be great if a comment could be added to the code so that then the code is clear for future readers.  Thanks! :)